packages/model-viewer/src/three-components/Renderer.ts

```diff
@@ -55,15 +56,27 @@ const DEFAULT_LAST_STEP = 3;
  * the texture.
  */
 export class Renderer extends EventDispatcher {
-  private static _singleton = new Renderer({debug: isDebugMode()});
+  private static _singleton =
+      new Renderer({powerPreference: 'high-performance', debug: isDebugMode()});
 
   static get singleton() {
     return this._singleton;
   }
 
   static resetSingleton() {
-    this._singleton.dispose();
-    this._singleton = new Renderer({debug: isDebugMode()});
+    const elements = this._singleton.dispose();
+    for (const element of elements) {
+      element.disconnectedCallback();
+    }
+
+    this._singleton = new Renderer({
+      powerPreference: ModelViewerElementBase.powerPreference,
+      debug: isDebugMode()
+    });
+
+    for (const element of elements) {
+      element.connectedCallback();
```

Comment:
disconnecting any existing model-viewer element from the old singleton and reconnecting them to the new one was all that was required to get the state back in shape. If they already had a loaded model, I'd guess it'll still be broken until they reset their `src`, but I think I'm okay with that, as it'll serve as a reminder that they really shouldn't be resetting after they have models loaded.